presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -408,10 +409,10 @@ public Page getOutput()
             outputIterator = null;
 
             if (finishing) {
-                if (!inputProcessed && !step.isOutputPartial()) {
+                if (!inputProcessed && (step.isOutputPartial() || step.equals(SINGLE))) {
```

Comment:
I have rearranged commits, and I refactor/introduce `produceDefaultOutput` before distributing ROLLUP.